include/kodlab_mjbots_sdk/limb_base.h

```diff
@@ -0,0 +1,155 @@
+/**
+ * @file limb_base.h
+ * @author Kodlab - Zac Gong (zacgong@seas.upenn.edu)
+ * @brief Abstract class for limbs, consisting of multiple joints
+ * @version 0.1
+ * @date 2022-08-18
+ *
+ * @copyright BSD 3-Clause License, Copyright (c) 2021 The Trustees of the University of Pennsylvania. All Rights Reserved
+ *
+ */
+
+#pragma once
+#include <limits>
+#include <memory>
+#include <vector>
+#include <type_traits>
+#include <string>
+#include "kodlab_mjbots_sdk/joint_base.h"
+#include <Eigen/Dense>
+
+namespace kodlab
+{
+    /**
+     * Abstract limb class that takes in a series of joints and treats it as an abstract limb
+     * Contains the kinematics and configurations for a leg
+     * @brief Abstract base class for legs consisting of multiple joints to be used in the robot class
+     */
+
+    struct LimbConfig
+    {
+        std::vector<std::array<float, 3>> actuator_offsets; // Offset between centers of two joints, in the frame of the first joint
+        std::vector<Eigen::Matrix3f> actuator_orientations; // Rotation matrices for each joint with respect to the previous joint (begins with R0_1)
+        Eigen::MatrixXf selection_matrix;
+    };
+
+    class LimbBase
+    {
+    public:
+        /**
+         * @brief Construct a new Limb Base object with a name
+         *
+         * @param name          /// Sets the limb name
+         * @param joints        /// Joints that make up the leg
+         * @param config        /// Configuration of the joints in the leg
+         *
+         */
+
+        LimbBase(
+            std::string name,
+            std::vector<std::shared_ptr<JointBase>> joints,
+            LimbConfig config);
+
+        /**
+         * @brief Construct a new Limb Base object
+         *
+         * @param joints        /// Joints that make up the leg
+         * @param config        /// Configuration of the joints in the leg
+         *
+         */
+
+        LimbBase(
+            std::vector<std::shared_ptr<JointBase>> joints,
+            LimbConfig config);
+
+        virtual ~LimbBase(){};
+
+        /**
+         * @brief Update position and velocity of each joint within leg
+         *
+         * @param pos_list desired positions of each joint
+         * @param vel_list desired velocities of each joint
+         *
+         */
+        void Update(std::vector<float> pos_list, std::vector<float> vel_list);
```

Comment:
Pass moderate cost-to-move input parameters by constant reference (see [here](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f15-prefer-simple-and-conventional-ways-of-passing-information)), i.e., 
```suggestion
        void Update(const std::vector<float> &pos_list, 
                    const std::vector<float> &vel_list);
```

Reply:
fixed for larger formatting commit